spec/models/tag_spec.rb

```diff
@@ -522,4 +522,107 @@
       end
     end
   end
+
+  describe "canonical" do
+    let!(:work) { create(:work, fandom_string: fandom.name) }
+
+    context "when the tag is non-canonical" do
+      let(:fandom) { create(:fandom, canonical: false) }
+
+      it "making it canonical adds filters to tagged works" do
+        fandom.attributes = { canonical: true }
+        fandom.save!
+        expect(fandom.filtered_works.reload).to contain_exactly(work)
+      end
+    end
+
+    context "when the tag is canonical" do
+      let(:fandom) { create(:fandom, canonical: true) }
+
+      it "making it non-canonical removes filters" do
+        fandom.attributes = { canonical: false }
+        fandom.save!
+        expect(fandom.filtered_works.reload).to contain_exactly()
+      end
+    end
+  end
+
+  describe "merger" do
+    let!(:work) { create(:work, fandom_string: synonym.name) }
+
+    let(:fandom) { create(:fandom, canonical: true) }
+    let(:synonym) { create(:fandom, canonical: false) }
+
+    context "when the tag has no synonym" do
+      it "adding a synonym adds filters" do
+        synonym.attributes = { syn_string: fandom.name }
+        synonym.save!
+        expect(fandom.filtered_works.reload).to contain_exactly(work)
+      end
+
+      context "when the tag is canonical" do
+        let(:synonym) { create(:fandom, canonical: true) }
+
+        it "adding a synonym changes the tag to non-canonical" do
+          synonym.attributes = { syn_string: fandom.name }
+          synonym.save!
+          expect(synonym.reload.canonical).to be_falsey
+        end
+
+        it "adding a synonym transfers its metatags and subtags" do
+          sub = create(:fandom, canonical: true)
+          meta = create(:fandom, canonical: true)
+          synonym.sub_tags << sub
+          synonym.meta_tags << meta
+          synonym.reload
+
+          synonym.attributes = { syn_string: fandom.name }
+          synonym.save!
+
+          expect(fandom.sub_tags.reload).to contain_exactly(sub)
+          expect(fandom.meta_tags.reload).to contain_exactly(meta)
+          expect(synonym.sub_tags.reload).to contain_exactly()
+          expect(synonym.meta_tags.reload).to contain_exactly()
+        end
+
+        it "adding a synonym transfers its associations", :pending do
```

Comment:
I wrote this test to make sure that I didn't break `add_merger_associations` by changing it from asynchronous to synchronous, but it turns out that this also fails on the master branch as well, since the Uncategorized Fandom media is added to the synonym fandom. I don't think that's the desired behavior, but I wanted to keep the scope of this pull request fairly small and not add in loads of extra changes.